src/DeepCopy/Reflection/ReflectionHelper.php

```diff
@@ -21,16 +21,19 @@ public static function getProperties(\ReflectionClass $ref)
         $propsArr = array();
 
         foreach ($props as $prop) {
-            $f = $prop->getName();
-            $propsArr[$f] = $prop;
+            $propertyName = $prop->getName();
+            $propsArr[$propertyName] = $prop;
         }
 
         if ($parentClass = $ref->getParentClass()) {
             $parentPropsArr = self::getProperties($parentClass);
-            if (count($parentPropsArr) > 0) {
-                $propsArr = array_merge($parentPropsArr, $propsArr);
+            foreach ($propsArr as $key => $property) {
+                $parentPropsArr[$key] = $property;
             }
```

Comment:
No, the test only assert that `$a3` is in the properties list returned by the helper.
What you have to test, and then to ensure, is that the `ReflectionProperty` returned refers to the `$a3` of `ReflectionHelperTestChild` instead of `ReflectionHelperTestParent` one.